src/Http/RuleMatcher.php

```diff
@@ -30,66 +30,35 @@ class RuleMatcher implements RuleMatcherInterface
     private $requestMatcher;
 
     /**
-     * @var array
+     * @var ResponseMatcherInterface
      */
-    private $criteria;
+    private $responseMatcher;
 
     /**
-     * @var ExpressionLanguage
+     * @param RequestMatcherInterface  $requestMatcher  Strategy to match the request
+     * @param ResponseMatcherInterface $responseMatcher Strategy to match the response
      */
-    private $expressionLanguage;
-
-    /**
-     * @param RequestMatcherInterface $requestMatcher Strategy to match the request
-     * @param array                   $criteria       Additional criteria not covered by request matcher
-     */
-    public function __construct(RequestMatcherInterface $requestMatcher, array $criteria)
-    {
+    public function __construct(
```

Comment:
please add a method doc that explains that both matchers must match for the rule matcher to match.

and i wonder if RuleMatcher is still the right name for this. its actually a RequestResponseMatcher or a HttpMatcher. not happy with both those names, but well... and we do a BC break here anyways, might as well rename it if we find a better name.